### ISSUE TITLE
How to have a "save draft" use a different validate function?

### ISSUE THREAD

**Issue by gnarf** [erikras/redux-form#2511]
I'm curious if anyone has any examples of using redux-form to have validation for the form, but allowing a "save draft" type button that would allow submitting with a different set of validation rules.

I.E. Some fields can be empty in draft state, but should be forced to be filled out in order to use the "Start" submit button.

**Comment by erikras** [comment]
Seems like that could be done via [submit validation](http://redux-form.com/6.4.3/examples/submitValidation/), no?

**Comment by gnarf** [comment]
Sure! I was thinking this might actually make a good documentation example, which is one of the reasons I was opening an issue also.

I ended up doing something with submit validation, but now I've lost any sync validation capabilities.

I still think some way to allow a button to be bound to attempt to submit even though errors are present ala `handleSubmit(saveDraft, { allowInvalid: true })` might be a useful future feature.

Or even tying it in with `shouldValidate()` somehow, passing a flag saying which submit handler is being used?

I'd really love to be able to have the sync validation errors showing, but still allow people to "save draft"

**Comment by gnarf** [comment]
> Meanwhile, StackOverflow is better at keeping popular questions visible, while separating feature-related questions from help-related questions.

I totally get that [:smile:](https://stackoverflow.com/u/91914) -- You might want to update the website to not link to stack overflow or issue trackers next to "Ask for help" as I interpreted this as you would welcome questions in either forum. 
 
![image](https://cloud.githubusercontent.com/assets/549355/22339864/807e0698-e3b9-11e6-97eb-4592f1a7bc0d.png)

**Comment by gustavohenke** [comment]
You can always ping us here or at Twitter with the question you need help with if you're not getting answers :)

Also, I'm afraid that the alert GitHub shows pointing to `CONTRIBUTING.md` when you're about to open an issue makes no difference at all 😢

**Comment by gnarf** [comment]
Anyway -- issue at hand.

Would there be a way to tell `redux-form` to use a sync/async validate method for displaying errors, but to flag that my submit functions will do their own error check on submit?

Even a simple flag like `reduxForm({ allowSubmitWhileInvalid: true })` (name bike-shedding seems appropriate ;) )

**Comment by rizedr** [comment]
@gustavohenke & @erikras, I agree with @gnarf, this is an important use case of form validation that would be cool if it were properly covered by reduxForm.

Let's say that you are building a CMS interface. In it you'd have the ability to create posts. When you are creating a post you have 2 actions: `Save as a draft` or `Publish`.

When saving as a drafts, you'd be interested to validate that the title is filled. On the other hand, when publishing, you'd be interested to validate that both the title and body are field.

I agree that this is totally achievable with the submit validation example, however, what if you are working on an accounting CMS (real world example) and your post all of the sudden has 20 fields that require validation as a draft and 80 in total when publishing.

Therefore, in my opinion, a flag passed at submission level would be a nice future, because afterwards you could tailor your field validators after that. Take this as an example:

```class Example extends React.Component {

  validatePost = (validators) => {
    // get submissionType from somewhere
    if (submissionType === 'post') {
      return validators;
    }

    return undefined;
  };

  validateDraft = (validators) => {
    // get submissionType from somewhere
    if (submissionType === 'draft') {
      return validators;
    }

    return undefined;
  };
  
  handlePost = () => {
    // set submissionType to post,
    // validate,
    // if valid continue submission
  };
  
  handleSaveDraft = () => {
    // set submissionType to draft,
    // validate,
    // if valid continue submission
  };

  render() {

    return (
      <form>
        {/* validate on draft & post */}
        <Field name="title" type="text" component="input" validate={[ this.validateDraft([required]), this.validatePost([required]) ]} />
        
        {/* validate on draft & post */}
        <Field name="shortDescription" type="text" component="input" validate={[ this.validateDraft([required]), this.validatePost([required]) ]} />
        
        {/* validate on post */}
        <Field name="longDescription" type="text" component="input" validate={[ this.validatePost([required]) ]} />

        <button onClick={this.handlePost}>Post</button>
        <button onClick={this.handleSaveDraft}>Save Draft</button>
      </form>
    );
  }
}


export default reduxForm({
  form: 'postForm'
})(Example);``` 

To sum up, I think this should be opened and labeled as an `enhancement`.

**Comment by shogan94** [comment]
Has anyone found a solution to doing this?
